ospd/vtfilter.py

```diff
@@ -0,0 +1,112 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+
+""" Vulnerability Test Filter class.
+"""
+import re
+import operator
+
+class VtsFilter(object):
+    """ Helper class to filter Vulnerability Tests """
+
+    def __init__(self):
+        """ Initialize filter operator and allowed filters. """
+        self.filter_operator = {
+            '<': operator.lt,
+            '>': operator.gt,
+            '=': operator.eq,
+        }
+
+        self.allowed_filter = {
+            'creation_time': self.format_vt_creation_time,
+            'modification_time': self.format_vt_modification_time,
+        }
+
+    def parse_filters(self, vt_filter):
+        """ Parse a string containing one or more filters
+        and return a list of filters
+
+        Arguments:
+            vt_filter (string): String containing filters separated with
+                semicolon.
+        Return:
+            List with filters. Each filters is a list with 3 elements
+            e.g. [arg, operator, value]
+        """
+
+        filter_list = vt_filter.split(';')
+        filters = list()
+        for single_filter in filter_list:
+           filters.append(re.split('(\W)', single_filter, 1))
+
+        for _element, _oper, _val in filters:
+            if _element not in self.allowed_filter or _oper not in self.filter_operator:
+                filters.remove([_element, _oper, _val])
+
+        return filters
+
+    def format_vt_creation_time(self, value):
+        """ In case the given creationdatetime value must be formated,
+        this function must be implemented by the wrapper
+        """
+        return value
+
+    def format_vt_modification_time(self, value):
+        """ In case the given modification datetime value must be formated,
+        this function must be implemented by the wrapper
+        """
+        return value
+
+    def format_filter_value(self, element, value):
+        """ Calls the specific function to format value,
+        depending on the given element.
+        """
+        format_func = self.allowed_filter.get(element)
+        return format_func(value)
+
+    def get_filtered_vts_list(self, vts, vt_filter=None):
+        """ Gets a collection of vulnerability test from the vts dictionary,
+        which match the filter.
+
+        Arguments:
+            vt_filter (string): Filter to apply to the vts collection.
+
+        Returns:
+            Dictionary with filtered vulnerability tests.
+        """
+        if not vt_filter:
+            raise RequiredArgument('vt_filter: A valid filter is required.')
+
+        filters = self.parse_filters(vt_filter)
+        if not filters:
+            return None
+
+        _vts_aux = vts.copy()
+        for _element, _oper, _filter_val in filters:
+            for vt_id in _vts_aux.copy():
```

Comment:
Maybe it is not necessary to create another copy by using? Not sure.
```suggestion
            for vt_id in vts:
```

Reply:
The copy is need since the _vts_aux dictionary is change during the iteration.

Reply:
Yeah but _vts_aux is already a copy of vts. Therefore IMHO it should be possible to iterate over the original unchanged vts variable.

Reply:
At this moment you can pass more than one filter option separated by a semicolon. E.g. : 
`<get_vts filter='modification_time&gt;201903130813;modification_time&lt;201903130815'/>`

The second one must be applied over the result of the first one (logical AND operation). 
If vts is used instead, the logical operation will be an OR, and the result will be the whole vts collection.